test/controllers/concerns/authentication_test.rb

```diff
@@ -172,6 +176,35 @@ def raising
       assert_not User.current, 'current user must be reset after request'
     end
 
+    should 'account found, is_internal unset, updates it' do
+      FactoryBot.create(:account, account_number: '1234')
+      assert_nil Account.find_by(account_number: '1234').is_internal
+
+      encoded_header = Base64.encode64(
+        {
+          'identity':
+          {
+            'account_number': '1234',
+            'user': {
+              'username': 'username',
+              'is_internal': false
+            }
+          },
+          'entitlements':
+          {
+            'insights': {
+              'is_entitled': true
+            }
+          }
+        }.to_json
+      )
+
+      process_test(headers: { 'X-RH-IDENTITY': encoded_header })
+      assert_response :success
+      assert_equal '1234', response.body
+      assert_equal false, Account.find_by(account_number: '1234').is_internal
```

Comment:
I like to use `refute`, but rubocop doesn't like it:

```suggestion
      assert_not Account.find_by(account_number: '1234').is_internal
```

Reply:
`assert_not` also accepts `nil` which we don't want